File: bonus.go
package main

type Bonus struct {
	*Ghost1
	lifeforce int
	typ       int
	alive     bool
}

func newBonus(filename string, x, y, speed int, af *AnimationFactory, typ int) *Bonus {
	return &Bonus{
		Ghost1:    newGhost1(filename, x, y, speed, af),
		lifeforce: 300,
		alive:     true,
		typ:       typ,
	}
}

func (b *Bonus) decLife() {
	if b.lifeforce > 0 {
		b.lifeforce--
	}
}

File: orig/hhh-2008-03-29/Bonus.cc
#include "Bonus.h"

class Bonus;

Bonus::Bonus(char *filename, int x, int y, int speed,AnimationFactory* af, char type):
	Ghost1(filename, x, y, speed, af){
	this->lifeforce = 300;
	this->alive = true;
	this->type = type;
};

void Bonus::decLife(){
	if(lifeforce>0) lifeforce--;
};
//Food::~Food(){
//	int i;
//	for(i=0; i<frames.size(); i++){
//		SDL_FreeSurface(frames[i]);
//	}
//};
